homeassistant/components/smartthings/sensor.py

```diff
@@ -360,3 +365,61 @@ def state(self):
             return three_axis[self._index]
         except (TypeError, IndexError):
             return None
+
+
+class SmartThingsPowerConsumptionReportSensor(SmartThingsEntity):
+    """Define a SmartThings Power Consumption Report."""
+
+    def __init__(self, device):
+        """Init the class."""
+        super().__init__(device)
+        self._attribute = Attribute.power_consumption
+        self._name = "Power Consumption Report"
+        self._primary_attribute = "power_consumption_energy"
+        self._device_class = DEVICE_CLASS_ENERGY
+        self._default_unit = ENERGY_WATT_HOUR
+
+    @property
+    def name(self) -> str:
+        """Return the name of the binary sensor."""
+        return f"{self._device.label} {self._name}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{self._device.device_id}.{self._attribute}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return getattr(self._device.status, self._primary_attribute)
```

Comment:
This is unnecessary abstraction. Simply return `self._device.status.power_consumption_energy`

Reply:
https://github.com/home-assistant/core/pull/43062/commits/4f7ca579a4d51fd7926ac0b4b8f92f99f243f3b0#diff-9151abfb874472904c2e6b05b99f2650845d7cec0ded2a73a4950c1708df04ffR388-L396